libraries/botframework-connector/src/auth/passwordServiceClientCredentialFactory.ts

```diff
@@ -68,45 +68,22 @@ export class PasswordServiceClientCredentialFactory implements ServiceClientCred
         }
 
         let credentials: MicrosoftAppCredentials;
-        let normalizedEndpoint = loginEndpoint?.toLowerCase();
+        const normalizedEndpoint = loginEndpoint?.toLowerCase();
+
         if (normalizedEndpoint?.startsWith(AuthenticationConstants.ToChannelFromBotLoginUrlPrefix)) {
-            // TODO: Unpack necessity of these empty credentials based on the loginEndpoint as no tokens are fetched when auth is disabled.
-            credentials =
-                appId == null
-                    ? MicrosoftAppCredentials.Empty
-                    : new MicrosoftAppCredentials(appId, this.password, this.tenantId, audience);
+            credentials = new MicrosoftAppCredentials(appId, this.password, this.tenantId, audience);
         } else if (normalizedEndpoint === GovernmentConstants.ToChannelFromBotLoginUrl.toLowerCase()) {
-            credentials =
-                appId == null
-                    ? new MicrosoftAppCredentials(
-                          undefined,
-                          undefined,
-                          undefined,
-                          GovernmentConstants.ToChannelFromBotOAuthScope
-                      )
-                    : new MicrosoftAppCredentials(appId, this.password, this.tenantId, audience);
-            normalizedEndpoint = loginEndpoint;
+            credentials = new MicrosoftAppCredentials(appId, this.password, this.tenantId, audience);
         } else {
-            credentials =
-                appId == null
-                    ? new PrivateCloudAppCredentials(
-                          undefined,
-                          undefined,
-                          undefined,
-                          undefined,
-                          normalizedEndpoint,
-                          validateAuthority
-                      )
-                    : new PrivateCloudAppCredentials(
-                          appId,
-                          this.password,
-                          this.tenantId,
-                          audience,
-                          normalizedEndpoint,
-                          validateAuthority
-                      );
+            credentials = new PrivateCloudAppCredentials(
+                appId,
```

Comment:
If `appId` is `undefined`, are `password` and `tenantId` also undefined? Along with `audience`? I'm wary of this change.

Reply:
This change comes straight from porting the dotnet version, you can check it [here ](https://github.com/microsoft/botbuilder-dotnet/pull/5829/files?file-filters%5B%5D=.cs#diff-3bff06c07579e50f9e2f978a5b221eafc7f1f5ac5dff2af5eac6302a2637ed30). Besides that, this portion of code would not be reached because of the `if` on line 62, right?

Reply:
Ah fair - didn't realize the .NET piece was refactored.

Reply:
Yes, this was code I didn't have time to 100% test and clean up when we shipped the JS, Python and Java implementations of CloudAdapter. Without an AppId, it's impossible to fetch a token from AAD which is why the check on line 62 exists and an [issue was filed in .NET](https://github.com/microsoft/botbuilder-dotnet/issues/5591#issuecomment-864167043). Giving the bot a `MicrosoftAppCredentials.Empty` instance is functionally equivalent to giving it an empty `PrivateCloudAppCredentials` or a [`MicrosoftGovernmentAppCredentials.Empty`](https://github.com/microsoft/botbuilder-dotnet/pull/5730/files#diff-3bff06c07579e50f9e2f978a5b221eafc7f1f5ac5dff2af5eac6302a2637ed30L90) in .NET.